bigquery/storage/managedwriter/appendresult.go

```diff
@@ -66,7 +67,9 @@ func (ar *AppendResult) GetResult(ctx context.Context) (int64, error) {
 // append request.
 type pendingWrite struct {
 	request *storagepb.AppendRowsRequest
-	result  *AppendResult
+	// for schema evolution cases, accept a new schema
+	newSchema *descriptorpb.DescriptorProto
```

Comment:
Will this property need to be removed once 205756033 is resolved? It might be redundant with `AppendRowsRequest.proto_rows.writer_schema.proto_descriptor`.

Or am I understanding that email thread that we won't be exposing the underlying AppendRowsRequest to users?

Reply:
In this veneer I'm wrapping the append request.  My expectation is we'll end up with per-stream caches of schema and per-stream append queues for the multiplexing case.  Currently we just retain a single schema and append queue (though in go it's a channel rather than a queue).

Reply:
Is this also the reason why you can optionally have a new schema updated as part of a return error. I was puzzled how one can automatically fix schema updates on runtime though? Is there any realistic example of this?

Reply:
Yeah, the returned schema is the notification of schema extension.  It's a case where you do something like an ALTER TABLE ADD COLUMN or extend schema via tables.update while streaming data, and the change gets acknowledged by the streaming backend by setting the new schema.  My expectation is we'll add a callback registration for this as well, but not in this change.